Tasks/VsTest/vstest.ts

```diff
@@ -343,8 +350,249 @@ function getTestResultsDirectory(settingsFile: string, defaultResultsDirectory:
     return defer.promise;
 }
 
+function getTICollectorURI () {
+    return "datacollector://microsoft/TestImpact/1.0";
+}
+
+function getTIAssemblyQualifiedName(vsVersion) {
+    return "Microsoft.VisualStudio.TraceCollector.TestImpactDataCollector, Microsoft.VisualStudio.TraceCollector, Version=" + vsVersion + ".0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a";
+}
+
+function getTIFriendlyName() {
+    return "Test Impact";
+}
+
+function getTestImpactAttributes(vsVersion) {
+    return { uri: getTICollectorURI(),
+             assemblyQualifiedName: getTIAssemblyQualifiedName(vsVersion),
+             friendlyName: getTIFriendlyName() };
+}
+
+function isTestImapctCollectorPresent(dataCollectorArray) {
+    var found = false;
+    var tiaFriendlyName = getTIFriendlyName();
+    for (var i = 0; i < dataCollectorArray.length; i++) {
+        try {
+            if (dataCollectorArray[i].$.friendlyName === tiaFriendlyName) {
+                tl.debug("Test impact data collector already present, will not add the node.");
+                found = true;
+                break;
+            }
+        }
+        catch (e) {
+        }
+    }
+    return found;
+}
+
+function updateRunSettingsFileForTestImpact(vsVersion: number, settingsFile: string, exitErrorMessage:string): Q.Promise<string> {
+    var defer = Q.defer<string>();
+    tl.debug("Adding test impact data collector element to runsettings file provided.");
+    readFileContents(settingsFile, "utf-8")
+        .then(function (xmlContents) {
+            var parser = new xml2js.Parser();
+            parser.parseString(xmlContents, function (err, result) {
+                if (err) {
+                    tl.warning(tl.loc('ErrorWhileReadingRunSettings', err));
+                    tl.debug(exitErrorMessage);
+                    defer.resolve(settingsFile);
+                    return defer.promise;
+                }
+                if (result.RunSettings === undefined) {
+                    tl.warning(tl.loc('FailedToSetTestImpactCollectorRunSettings'));
+                    defer.resolve(settingsFile);
+                    return defer.promise;
```

Comment:
this error prints a warning on the console. the execution continues with the settings file unchanged.